content/en/apps/guides/hosting/core-developer.md

```diff
@@ -0,0 +1,240 @@
+. ~/.$0rc
+```
+
+In order for Docker to boot correctly, restart entire machine, which will complete the "Install" Section:
+
+```shell
+sudo reboot
+```
+
+### CouchDB Setup in Docker
+
+Before we get started, let's run the simple `hello-world` Docker container. This will ensure docker is working as expected and output "Hello from Docker!" as well as some other intro text:
+
+```shell
+docker run hello-world
+```
+
+Now that we know Docker is set up, let's start our CouchDB container. Note this will run in the background and store its data in `/home/YOUR-USER/cht-docker`. The login for your CHT instance will be `medic` and the password will be `password`:
+
+```shell
+docker run -d -p 5984:5984 -p 5986:5986 --name medic-couchdb -e COUCHDB_USER=medic -e COUCHDB_PASSWORD=password -v ~/cht-docker/local.d:/opt/couchdb/data -v ~/cht-docker/local.d:/opt/couchdb/etc/local.d apache/couchdb:2
+```
+
+Let's ensure CouchDB is set up with a test `curl` call. This should show "nonode@nohost" in JSON:
+
+```shell
+curl -X GET "http://medic:password@localhost:5984/_membership" | jq
+```
+
+Every time you run any `grunt` or `node` commands, it will expect `COUCH_NODE_NAME` and `COUCH_URL` environment variables to be set:
+
+```shell
+echo "export COUCH_NODE_NAME=nonode@nohost">> ~/.$0rc
+echo "export COUCH_URL=http://medic:password@localhost:5984/medic">> ~/.$0rc
+. ~/.$0rc
+echo $COUCH_NODE_NAME && echo $COUCH_URL
+```
+
+### CHT Core Cloning and Setup
+
+Clone the main CHT Core repo from GitHub and change directories into it:
+
+```shell
+git clone https://github.com/medic/cht-core ~/cht-core
+cd ~/cht-core
+```
+
+Install dependencies and perform other setup tasks via an `npm` command. Note this command may take many minutes. Be patient!
+
+```shell
+npm ci
+```
+
+We need to harden couch with a `grunt` call, required even in development:
+
+```shell
+grunt secure-couchdb
+curl -X PUT "http://medic:password@localhost:5984/_node/$COUCH_NODE_NAME/_config/httpd/WWW-Authenticate" -d '"Basic realm=\"administrator\""' -H "Content-Type: application/json"
+```
+
+### Developing
+
+Now you have everything installed and can begin development! You'll need three separate terminals when doing development. In the first terminal, run:
+
+```shell
+cd ~/cht-core && grunt
+```
+
+Be **very** patient until you see:
+
+> "Waiting..."
+
+Then in a 2nd terminal run:
+
+```shell
+cd ~/cht-core && grunt dev-api
+```
+
+Finally, in a 3rd terminal run:
+
+```shell
+cd ~/cht-core && grunt dev-sentinel
+```
+
+That's it!  Now when you edit code in your IDE, it will automatically reload.  You can see the CHT running locally here: [http://localhost:5988/](http://localhost:5988/)
+
+When you're done with development you can `ctrl + c` in the three terminals and stop the CouchDB container with `docker stop medic-couchdb`.  When you want to resume development later, run `docker start medic-couchdb` and re-run the three terminal commands.
+
+## Other Paths
```

Comment:
Maybe call this "Troubleshooting"? If you've forgotten (like I had) that the first heading was "Happy Path" this heading makes no sense.

Reply:
renamed AND linked - let's help find the right path! see 5a7ae9a